server/resp/src/main/java/org/infinispan/server/resp/operation/LCSOperation.java

```diff
@@ -36,6 +46,16 @@ public static CompletionStage<LCSResponse> performOperation(AdvancedCache<byte[]
             });
    }
 
+   private static byte[] findValue(Map<byte[], byte[]> values, byte[] key) {
+      assert values.size() == 2;
+      for (Map.Entry<byte[], byte[]> entry : values.entrySet()) {
```

Comment:
Was this working before?? The hashCode and equals for an array only works for the same instance... so I can't see how it was. Sounds like this might be a separate bug even?

Reply:
I caught this with the simple cache. Before that, the tests were not failing. I guess it just didn't trigger the failure.